src/components/button-minor/button-minor.test.js

```diff
@@ -0,0 +1,461 @@
+    });
+
+    it.each([
+      [BUTTON_ICON_POSITIONS[0], "right"],
+      [BUTTON_ICON_POSITIONS[1], "left"],
+    ])(
+      "should set position to %s for icon in a button",
+      (iconPosition, margin) => {
+        CypressMountWithProviders(
+          <ButtonMinor iconType="add" iconPosition={iconPosition}>
+            IconPosition
+          </ButtonMinor>
+        );
+
+        icon().should("have.css", `margin-${margin}`, "8px");
+      }
+    );
+
+    it("should render Button Minor with full width", () => {
+      CypressMountWithProviders(<ButtonMinor fullWidth>Foo</ButtonMinor>);
+
+      buttonDataComponent().then(($el) => {
+        useJQueryCssValueAndAssert($el, "width", 1365);
+      });
+    });
+
+    it("should render Button Minor with href", () => {
+      CypressMountWithProviders(
+        <ButtonMinor href="https://carbon.sage.com/">Foo</ButtonMinor>
+      );
+
+      buttonDataComponent().should(
+        "have.attr",
+        "href",
+        "https://carbon.sage.com/"
+      );
+    });
+
+    it.each([
+      ["true", "white-space"],
+      ["false", "flex-wrap"],
+    ])(
+      "should render the Button Minor text without wrapping when noWrap is called",
+      (booleanState, cssValue) => {
+        CypressMountWithProviders(
+          <ButtonMinor noWrap={booleanState}>
+            {" "}
+            Long long long long long text{" "}
+          </ButtonMinor>
+        );
+
+        if (booleanState === true) {
+          buttonDataComponent()
+            .children()
+            .children()
+            .should("have.value", cssValue, "wrap");
+        } else {
+          buttonDataComponent()
+            .children()
+            .children()
+            .should("not.have.value", cssValue, "wrap");
+        }
+      }
+    );
+
+    it("should check Button Minor is disabled", () => {
+      CypressMountWithProviders(<ButtonMinorDifferentTypes disabled />);
+
+      buttonDataComponent()
+        .eq(positionOfElement("first"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+      buttonDataComponent()
+        .eq(positionOfElement("second"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+      buttonDataComponent()
+        .eq(positionOfElement("third"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it("should check Button Minor is enabled", () => {
+      CypressMountWithProviders(<ButtonMinorDifferentTypes />);
+
+      buttonDataComponent().eq(positionOfElement("first")).should("be.enabled");
+      buttonDataComponent()
+        .eq(positionOfElement("second"))
+        .should("be.enabled");
+      buttonDataComponent().eq(positionOfElement("third")).should("be.enabled");
+    });
+
+    it("should check Button Minor is destructive", () => {
+      CypressMountWithProviders(<ButtonMinorDifferentTypes destructive />);
+
+      buttonDataComponent()
```

Comment:
we can put this in a `loop` and just avoid `first`, `second` and `third` and as well a code duplications

Reply:
In this test, first, second and third have different expected values. Some are transparent, some are destructive around the outline or background depending on the button type. Looping around this would be complicated.